src/solver/ia_main.jl

```diff
@@ -0,0 +1,250 @@
+using Symbolics
+E = Base.MathConstants.e
+
+function isolate(lhs, var)
+    rhs = Vector{Any}([0])
+    lhs = unwrap(lhs)
+    while !isequal(lhs, var)
+        subs, poly = filter_poly(lhs, var)
+        try
+            if check_poly_inunivar(poly, var)
```

Comment:
I think a try/catch would not be needed if `check_poly_inunivar` returned a boolean ?

Reply:
I think i had it there because if the poly contains one of our special functions such as slog, ssqrt, and scbrt, polynomial_coeffs throws an error, so i want to continue isolating.